packages/key-did-resolver/README.md

```diff
@@ -6,6 +6,325 @@
 ```
 $ npm install key-did-resolver
 ```
+### Usage
+This code includes support for Ed25519 and Secp256k1 did:keys described in https://github.com/w3c-ccg/did-method-key .
+
+Example code for creating a did:key for Ed25519 is here:
+https://github.com/ceramicnetwork/key-did-provider-ed25519
+
+Example code for creating a did:key for Secp256k1 is here:
+https://github.com/ceramicnetwork/key-did-provider-secp256k1
+
+This code supports compressed forms of P-256, P-384, and P-521 did:keys in concordance with https://github.com/w3c-ccg/did-method-key test-vectors.
+
+The did:key id is the multicodec name + the compressed public key expressed as a Uint8Array (byte array) then encoded with base58btc and expressed as a string.
+The syntax of a did:key is did:key:id.
+
+Compressed keys are the X coordinate of the public key with a prefix that depends on the sign of the Y curve coordinate. The prefix is '02' if even and '03' if odd.
+
+Example code for creating a did:key for P-256, P-384, and P-521 is here:
+https://github.com/bshambaugh/did-key-creator 
+
+## Code
+```
+// Usage from cloned GitHub Repository:
+// import * as keyDIDResolver from '../js-ceramic/packages/key-did-resolver/lib/index.js';
+
+import keyDIDResolver from 'key-did-resolver'
+
+const resolverRegistry = keyDIDResolver.getResolver();
+
+const resolve = resolverRegistry.key;
+
```

Comment:
Can you make the example more like this?
https://www.npmjs.com/package/@ceramicnetwork/3id-did-resolver

i.e. show it's usage together with the `did-resolver` pkg

Reply:
import {DID} from 'dids'
import KeyResolver from 'key-did-resolver'
import {Ed25519Provider} from 'key-did-provider-ed25519'
import {randomBytes} from '@stablelib/random'

// this stuff might be a bit beyond the scope
import { CeramicClient } from '@ceramicnetwork/http-client'
const API_URL = "https://ceramic-clay.3boxlabs.com"
const ceramic = new CeramicClient(API_URL)
//

const seed = randomBytes(32) //  32 bytes with high entropy
const provider = new Ed25519Provider(seed)

const did = new DID({ provider, resolver: KeyResolver.getResolver() })
const auth = await did.authenticate()
console.log('auth is')
console.log(auth) // this spits out the did:key

// create JWS
const { jws, linkedBlock } = await did.createDagJWS({ hello: 'world' })

// create JWE
const jwe = await did.createDagJWE({ very: 'secret' }, [did.id])

console.log('jwe is:');
console.log(jwe);

// decrypt JWE
const decrypted = await did.decryptDagJWE(jwe)

console.log('decrypted is:');
console.log(decrypted);

// this stuff may be a bit beyold the scope
ceramic.did = did
ceramic.did.setProvider(provider)
//

Reply:
That's my latest draft.

Reply:
See the README for the latest. I think I was headed down the wrong path originally, but it made me think about did-jwt, did-jwt-vc, and future directions I may want to take https://github.com/bshambaugh/did-key-creator .